src/System.Collections.Concurrent/src/System/Collections/Concurrent/ConcurrentDictionary.cs

```diff
@@ -1207,27 +1207,45 @@ public bool IsEmpty
         {
             get
             {
+                // Check if any buckets are non-empty, without acquiring any locks.
+                // This fast path should generally suffice as collections are usually not empty.
```

Comment:
FWIW, I agree it should suffice though I'm not sure this is the reason. If it's not empty, the fast path will succeed.  If it is empty, then most likely there aren't a lot of threads mutating the dictionary, in which case the slow path won't have much contention in acquiring the locks.